### ISSUE TITLE
Linter fails to work properly when run with diff-quality

### ISSUE THREAD

**Issue by lisa-nalyvaiko** [sqlfluff/sqlfluff#4130]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I'm looking to lint only modified or new files, thus I'm trying to user diff-quality. However, when I run it with `diff-quality --violations sqlfluff --compare-branch=master --fail-under=100`, the linter isn't working properly. I get this:
```
Total:   55 lines
Violations: 0 lines
% Quality: 100%
```

When in reality there are in fact violations, which are caught by running `sqlfluff lint`.

### Expected Behaviour

Linter catches violations properly when being run with `diff-quality` command.

### Observed Behaviour

Linter misses code issues when being run with `diff-quality`.

### How to reproduce

- Have SQL codestyle issues
- Run `diff-quality --violations sqlfluff --compare-branch=master --fail-under=100`

### Dialect

Postgres

### Version

0.8.1

### Configuration

[sqlfluff]
dialect = postgres
exclude_rules = L009,L016,L031,L034,L044
# L009 - newline at end of file - dbt seems to remove those trailing newlines
# L016 - line too long - compiled lines end up longer than templated lines
# L031 - Avoid table aliases - aliases in general are ok
# L034 - wildcards, simple, calc field ordering - currently too many reorderings to satsify
# L044 - SELECT * is widely used across this repo

[sqlfluff:rules]
max_line_length = 80
comma_style = leading

# sql keywords should upper cased
[sqlfluff:rules:L010]
capitalisation_policy: upper

# sql identifiers should be lower cased
[sqlfluff:rules:L014]
extended_capitalisation_policy: lower
unquoted_identifiers_policy: all

# sql functions should upper cased
[sqlfluff:rules:L030]
capitalisation_policy: upper

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
diff-quality only checks the new or added lines, not the entire file.

Can you provide a small test git repo that reproduces the problem? Otherwise, we probably can't fix your issue.

**Comment by barrywhart** [comment]
I notice you are using SQLFluff 0.8.1. That version was released more than a year ago. Can you test with the latest version, 1.4.2?

**Comment by barrywhart** [comment]
I was testing `diff-quality` today with this repo, and it worked fine. 
[sql.tgz](https://github.com/sqlfluff/sqlfluff/files/10137106/sql.tgz)

When I run:
```
git checkout my_branch
diff-quality --violations sqlfluff --compare-branch=master --fail-under=100
```

I get a lot of output, and the last few lines are:
```
-------------
Total:   215 lines
Violations: 183 lines
% Quality: 14%
-------------
```

**Comment by lisa-nalyvaiko** [comment]
@barrywhart I've created a test repo, installed `sqlfluff==0.8.1` and `diff-quality --violations sqlfluff --compare-branch=master` works perfectly fine there. Any pointers as what might be causing the issue in my actual repo?

**Comment by barrywhart** [comment]
Is it possible you have a different version of the `diff-cover` package installed?

With Python environments (and software environments in general), there are generally two ways to try and ensure consistent behavior:
* Pin the versions of **everything** you're installing, not just the visible things. With Python, `pip-compile` or `poetry` are common ways to do this.
* Rebuild the environment frequently (i.e  destroy your virtualenv or uninstall everything, then reinstall everything). If you aren't pinning everything, frequent rebuilds mean that you're running the latest versions of things. This is not as good as pinning, because updating to new versions can introduce issues, but it does help ensure _some_ consistency across environments.

**Comment by lisa-nalyvaiko** [comment]
I've rebuilt my environment a few times, and linter with diff-quality started working properly! Although now I face the same issue, but in CI pipeline. When running `diff-quality --violations sqlfluff --compare-branch=master --fail-under=100` locally after the commit, I get correct output:
```
Total:   2 lines
Violations: 2 lines
% Quality: 0%
```

But when running it in CI, diff-quality acknowledges 2 changed lines, but shows 0 violations. 
For the context, in my pipeline I'm doing `git fetch --depth 50 origin master:master` before `diff-quality --violations sqlfluff --compare-branch=master --fail-under=100`
What could be the issue there?

**Comment by barrywhart** [comment]
Can you provide a sample git repo, or reproduce the issue with the repo I provided?

Without a specific test case, I can only make uninformed guesses.

* Try running a similar "git diff" command in your CI pipeline as the command diff-quality itself runs.
* 2 lines is a small change. Try making a large change to see if that shows some results.
* Verify you have the same versions of SQLFluff a d diff-quality installed in CI as your local environment.

**Comment by lisa-nalyvaiko** [comment]
It turned out I did have a different diff-cover versions installed! Keeping them same have fixed the problem. Thank you for your help!

**Comment by barrywhart** [comment]
Is it okay to close this issue?
